frame/elections-phragmen/src/lib.rs

```diff
@@ -653,9 +657,13 @@ decl_module! {
 
 impl<T: Config> Module<T> {
 	/// The deposit value of `count` votes.
-	fn deposit_of(count: usize) -> BalanceOf<T> {
-		T::VotingBondBase::get().saturating_add(
-			T::VotingBondFactor::get().saturating_mul((count as u32).into())
+	fn deposit_of(count: u32) -> BalanceOf<T> {
+		T::VotingBondStorageBase::get().saturating_add(
+			T::VotingBondStorageFactor::get().saturating_mul(count.into())
+		).saturating_add(
+			T::VotingBondWeightFactor::get().saturating_mul(
+				T::WeightInfo::election_phragmen(0, 1, count).saturated_into()
```

Comment:
it is a bit unual to me to pay some weight with some deposit.

But I think the number of vote is limited thus we can't use too much weight anyway isn't it ?

Reply:
Yes the deposit amount is limited. 

Indeed weight deposit is a new concept, but I think it makes sense. While you are imposing some computation on chain, you should at least reserve some stake for it.